### PR TITLE
Added in enhanced syntax

### PR DESCRIPTION
**Pull request by arigesher**
I found the current system too limiting for the rewriting that I wanted to do.  This patch implements an enhanced syntax that allows one to specify the exact regular expression replacement to be used.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -21,6 +21,7 @@
   },
   "scripts": {
     "prepublish": "./node_modules/coffee-script/bin/coffee -o lib/ src/",
+    "postinstall": "./node_modules/coffee-script/bin/coffee -o lib/ src/",
```
Comment (joscha) [inline package.json]: coffee is a devDependency, so postinstall won't work - is there any specific reason you added it?

**Comment by joscha** [comment]
nice work @arigesher! If you can add a test for it, I am happy to merge and make a new release!

**Comment by arigesher** [comment]
Test added!

File: package.json (hunk 1 above)
Comment (joscha) [inline package.json]: @arigesher could you remove this change, please?

**Comment by joscha** [comment]
Thanks, I just noticed however that the argument names are not consistent with https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/replace - I wonder if allowing a callback would be better/more useful. Also then we don't specifically need to cater for the fact that replace actually allows the first param to be a string and the second parameter to be a callback.

**Comment by joscha** [comment]
@arigesher please try the `adaptPath` method in version `0.0.6`

**Pull request closed without merging**